scripts/ast-patches/changeWixAuthElevateFunctionType.js

```diff
@@ -0,0 +1,102 @@
+const ts = require("typescript");
+
+function changeWixAuthElevateFunctionType(ast) {
+  const wixAuthModule = ast.statements.find(
+    statement =>
+      ts.isModuleDeclaration(statement) && statement.name.text === "wix-auth"
+  );
+
+  if (!wixAuthModule) {
+    return ast;
+  }
+
+  const elevateFunction = wixAuthModule.body.statements.find(
+    statement =>
+      ts.isFunctionDeclaration(statement) &&
+      statement.name.escapedText === "elevate"
+  );
+
+  if (!elevateFunction) {
+    return ast;
+  }
+
+  // ast code was added with the help of https://ts-ast-viewer.com/#
+  elevateFunction.typeParameters = [getFunctionTypeParameter()];
+  elevateFunction.parameters = [getFunctionParameter()];
+  elevateFunction.type = getFunctionReturnType();
+
+  return ast;
+}
+
+function getFunctionReturnType() {
+  return ts.factory.createFunctionTypeNode(
+    undefined,
```

Comment:
this code is not human readable, you either should refactor it or just write the definition hard-coded

Reply:
Changed it